### PR TITLE
Fix layout picture item test

### PR DESCRIPTION
**Pull request by jef-n**
* add resources to core tests
* [API] align QgsPropertyCollection::value() with ::valueAsSomething()

### CODE REVIEW

File: python/core/auto_generated/qgspropertycollection.sip.in [hunk 1]
```diff
@@ -86,7 +86,7 @@ Returns a matching property from the collection, if one exists.
 .. seealso:: :py:func:`hasProperty`
 %End
 
-    virtual QVariant value( int key, const QgsExpressionContext &context, const QVariant &defaultValue = QVariant() ) const = 0;
+    virtual QVariant value( int key, const QgsExpressionContext &context, const QVariant &defaultValue = QVariant(), bool *ok /Out/ = 0 ) const = 0;
```
Comment (nyalldawson) [inline python/core/auto_generated/qgspropertycollection.sip.in]: Isn't this an API break? Python code will get a tuple now where it used to get a single value

Comment (jef-n) [reply python/core/auto_generated/qgspropertycollection.sip.in]: yes.  I added SIP_PYARGREMOVE now.

Comment (jef-n) [reply python/core/auto_generated/qgspropertycollection.sip.in]: which doesn't work with overrides

Comment (nyalldawson) [reply python/core/auto_generated/qgspropertycollection.sip.in]: Add a v2 version?

**Comment by jef-n** [comment]
Oh well, this is getting out of hand - value() should have the `ok`.  `SIP_PYARGREMOVE` to keep to old api doesn't work for this as `value()` is overridden in subclasses, adding `valueAsVariant` with `PYNAME(value)` and the original interface with the new one with `PYNAME(valueV2)` somehow also doesn't work - both apparently still return a tuple.    Probably it would be cleaner to just remove or deprecate all the `valueAs*` variants here anyway and use those of the `property()` instead.  Closing this in favor of less invasive #31383.

**Comment by nyalldawson** [comment]
No problem! Note that the original feature still needs a test, and was missing the [feature] tag so the doc team will need advising @Delazj

**Pull request closed without merging**